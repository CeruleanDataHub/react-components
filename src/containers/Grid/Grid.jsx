import PropTypes from "prop-types";
import React from "react";
import { Cell, Grid as GridWrapper } from "styled-css-grid";

const Grid = ({ children }) => (
  <GridWrapper columns={children.length}>{children}</GridWrapper>
);

Grid.propTypes = {
  children: PropTypes.node
};

Grid.defaultProps = {
  children: ""
};

export { Grid, Cell };
