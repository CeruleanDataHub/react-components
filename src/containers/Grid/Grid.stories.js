import React from "react";

import { Cell, Grid } from "./Grid";

export default {
  title: "Grid",
  component: Grid
};

export const GridStory = () => (
  <Grid>
    <Cell>foo</Cell>
    <Cell>bar</Cell>
    <Cell>baz</Cell>
  </Grid>
);
