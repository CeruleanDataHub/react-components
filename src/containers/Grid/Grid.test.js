import React from "react";
import { mount } from "enzyme";

import { Cell, Grid } from "./Grid";

describe("Grid", () => {
  it("renders", () => {
    const component = mount(
      <Grid>
        <Cell>Some content</Cell>
      </Grid>
    );

    expect(component).toMatchHtmlSnapshot();
  });
});
