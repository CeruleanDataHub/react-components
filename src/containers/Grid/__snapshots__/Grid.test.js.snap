// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render grid 1`] = `
.c0 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(0,1fr);
  grid-gap: 8px;
}

<div
  className="c0"
>
  
</div>
`;

exports[`Grid should render grid with cell 1`] = `
.c1 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
}

.c0 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(12,1fr);
  grid-gap: 8px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Content
  </div>
</div>
`;
